api/catalog/api/utils/throttle.py

```diff
@@ -7,24 +8,98 @@
 
 log = logging.getLogger(__name__)
 
+class ThrottleExemption:
```

Comment:
Should this also be an [abstract base class](https://docs.python.org/3/library/abc.html#abc.ABC)?
```suggestion
class ThrottleExemption(abc.ABC):
```